packages/server/src/sdk/app/tables/getters.ts

```diff
@@ -12,17 +12,37 @@ import {
   TableResponse,
   TableSourceType,
   TableViewsResponse,
+  View,
+  ViewV2,
   FeatureFlag,
 } from "@budibase/types"
 import datasources from "../datasources"
 import sdk from "../../../sdk"
+import { ensureQueryUISet } from "../views/utils"
+import { isV2 } from "../views"
+
+function processView(view: ViewV2 | View) {
+  if (!isV2(view)) {
+    return
+  }
+  ensureQueryUISet(view)
+}
+
+function processViews(views: (ViewV2 | View)[]) {
+  for (const view of views) {
+    processView(view)
+  }
+}
 
 export async function processTable(table: Table): Promise<Table> {
   if (!table) {
     return table
   }
 
   table = { ...table }
+  if (table.views) {
+    processViews(Object.values(table.views))
```

Comment:
Could this be a pure function and to return the processed views instead?

Reply:
I didn't return anything because it modifies what's passed in. I tend to avoid copying unless it's necessary because lots of it puts pressure on the GC.